pkg/executor/wasm/executor.go

```diff
@@ -91,31 +67,31 @@ func (*Executor) GetResourceBidStrategy(context.Context) (bidstrategy.ResourceBi
 func (e *Executor) makeFsFromStorage(
 	ctx context.Context,
 	jobResultsDir string,
-	volumes map[*model.StorageSpec]storage.StorageVolume,
+	volumes []storage.PreparedStorage,
 	outputs []model.StorageSpec) (fs.FS, error) {
 	var err error
 	rootFs := mountfs.New()
 
-	for input, volume := range volumes {
+	for _, v := range volumes {
 		log.Ctx(ctx).Debug().
-			Str("input", input.Path).
-			Str("source", volume.Source).
+			Str("input", v.Spec.Path).
+			Str("source", v.Volume.Source).
 			Msg("Using input")
 
 		var stat os.FileInfo
-		stat, err = os.Stat(volume.Source)
+		stat, err = os.Stat(v.Volume.Source)
 		if err != nil {
 			return nil, err
 		}
 
 		var inputFs fs.FS
 		if stat.IsDir() {
-			inputFs = os.DirFS(volume.Source)
+			inputFs = os.DirFS(v.Volume.Source)
```

Comment:
I think @simonwo is working on the dir/file breakdown, so just adding this comment to let him know when he's back from vacation.